File: LeetCode/Python/Easy/Longest Palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        if len(s) < 2:
            return len(s)
        d = {}
        
        for i in s:
            if i not in d.keys():
                d[i] = 1
            else:
                d[i] += 1
        
        odd = False
        count = 0
        
        for k,v in d.items():
            if v != 0 and v % 2 == 0:
                count += v
            elif  v != 0:
                if not odd:
                    count += v
                    odd = True
                else:
                    count += v -1
        
        return count

File: LeetCode/C#/Easy/Longest Palindrome.cs
public class Solution {
    public int LongestPalindrome(string s) {
        var d = new Dictionary<char,int>();
        foreach(char c in s.ToCharArray()){
            if (d.ContainsKey(c)) {
                d[c] += 1;
            } else {
                d.Add(c, 1);
            }
        }
        
        int result = 0;
        bool odd = false;
        foreach(char c in d.Keys){
            if(d[c] % 2 == 0){
                result += d[c];
            }
            else{
                odd = true;
                result += d[c] - 1;
            } 
        }
        
        return odd ? result + 1 : result;
    }
}